src/runtime/handle.ts

```diff
@@ -147,11 +147,7 @@ export class Collection extends Handle {
     assert(this.canRead, '_notify should not be called for non-readable handles');
     switch (kind) {
       case 'sync':
-        try {
-          particle.onHandleSync(this, this._restore(details));
-        } catch (e) {
-          this.reportUserExceptionInHost(e, particle, 'onHandleSync');
-        }
+        particle.callOnHandleSync(this, this._restore(details), e => this.reportUserExceptionInHost(e, particle, 'onHandleSync'));
```

Comment:
Why don't these need awaits?

Reply:
I don't know! The whole method is sync and I'm scared to change it.

Reply:
Variable._notify has awaits on all the onHandle calls; I think it's just an oversight that Collection doesn't.